src/main/java/com/coveo/pushapiclient/PlatformClient.java

```diff
@@ -282,6 +282,45 @@ public HttpResponse<String> deleteDocument(String sourceId, String documentId, B
         return this.httpClient.send(request, HttpResponse.BodyHandlers.ofString());
     }
 
+    public HttpResponse<String> openStream(String sourceId) throws IOException, InterruptedException {
+        String[] headers = this.getHeaders(this.getAuthorizationHeader(), this.getContentTypeApplicationJSONHeader());
+        URI uri = URI.create(this.getBasePushURL() + String.format("/sources/%s/stream/open", sourceId));
```

Comment:
nit: couldn't the base url formatting be moved into String.format() as well?

Reply:
Since that's the same for all other methods, I would tackle that in a separate PR (LENS-875)... to not increase the scope of this PR.